Do not penalize subdomains if sp is present but not equal to reject

jsf9k opened the pull request:
According to [RFC7489](https://tools.ietf.org/html/rfc7489#section-6.3),
> "sp" will be ignored for DMARC records published on subdomains of Organizational Domains
due to the effect of the DMARC policy discovery mechanism

Therefore we have chosen not to penalize subdomains for `sp!=reject` when considering whether subdomains satisfy a DMARC subdomain policy of reject.  This issue was originally brought to light in NCATS JIRA ticket CYHYDEV-761.

See also cisagov/trustymail#116 and jsf9k/cyhy-reports#32.

jsf9k committed:
Bump version from 1.1.15 to 1.1.16

jsf9k committed:
Do not penalize for sp!=reject when considering subdomains

According to RFC7489, "'sp' will be ignored for DMARC records
published on subdomains of Organizational Domains due to the effect of
the DMARC policy discovery mechanism."  Therefore we have chosen not
to penalize for sp!=reject when considering whether subdomains satisfy
a DMARC subdomain policy of reject.

See here for more details:
https://tools.ietf.org/html/rfc7489#section-6.3

dav3r commented on report/generate_trustymail_report.py:
```diff
@@ -397,6 +397,7 @@ def __score_domain(self, domain):
         score['num_dmarc_failures'] = num_dmarc_failures
 
         if domain['live']:
+            score['is_base_domain'] = domain['is_base_domain']
```
Why did you add `is_base_domain` to the `score`, but then not use it?

jsf9k replied:
I was going to use it in the mustache file but ended up not needing it.  I didn't see any harm in leaving it in there.  Do you?

dav3r replied:
Nope, just was wondering why- that is good by me.

dav3r commented on report/generate_trustymail_report.py:
```diff
@@ -438,7 +439,15 @@ def __score_domain(self, domain):
+            #
+            # See here for more details:
+            # https://tools.ietf.org/html/rfc7489#section-6.3
+            if not domain['is_base_domain'] or (
```
Doesn't this logic allow a subdomain with invalid DMARC to get `valid_dmarc_subdomain_policy_reject = True`?  Maybe that's ok?  I'm not quite sure.

jsf9k replied:
I think you're right.  This should be:
```python
if score['valid_dmarc'] and (not domain['is_base_domain'] or domain['dmarc_subdomain_policy'] == 'reject'):
```

Does that look correct to you?

dav3r replied:
Yes, I think that is better.  Don't forget to make the corresponding changes in the Cyber Exposure scorecard logic.

jsf9k replied:
Good catch!
:eagle: :eye:

jsf9k replied:
Fixed in 6ad8b62.

dav3r reviewed (approved):
I think this looks good, but please take a look at my comments.

Props for remembering to also make this same change to the Cyber Exposure Scorecard! 🎉

jsf9k commented:
FWIW, my intent is to do a full test scan with this code in my staging environment next week.

jsf9k committed:
Correct a bug

@dav3r noticed that the existing logic allowed a subdomain with an
invalid DMARC record be declared as having a valid DMARC subdomain
policy of reject.  This is incorrect.  Even if the domain not a base
domain it must have a valid DMARC record to achieve this distinction.

dav3r reviewed (approved):
Looks great now.  Strong work as always @jsf9k 💪 💼
